astropy/uncertainty/core.py

```diff
@@ -281,6 +281,9 @@ def view(self, dtype=None, type=None):
             else:
                 raise ValueError('Cannot set just dtype for a Distribution.')
 
+        if issubclass(type, Distribution):
```

Comment:
What's the driver here relative to `Masked`?  Is a comment warranted?

Reply:
See #11210 - I'd hoped to rebase once that is in.